apps/backend/src/users/users.controller.ts

```diff
@@ -71,6 +76,27 @@ export class UsersController {
     return users.map((user) => new UserDto(user));
   }
 
+  @Post('check-password-complexity')
+  async checkPasswordComplexity(
+    @Body() passwordData: {password: string}
+  ): Promise<string[]> {
+    if (typeof passwordData.password !== 'string') {
+      return ['Password must be of type string'];
+    }
+    const errors: string[] = [];
+    checkLength(passwordData.password) &&
+      errors.push('Password must be at least 15 characters');
+    !hasClasses(passwordData.password) &&
+      errors.push(
+        'Password must contain a combination of lowercase letters, uppercase letters, numbers, and special characters'
+      );
+    !noRepeats(passwordData.password) &&
+      errors.push(
+        'Password must not contain 4 consecutive characters of the same character class'
+      );
```

Comment:
The body of this seems more suited for another function in the password complexity pipe. That way it is possible to just `errors.join(' ')` to generate the bad request message (prepending `Password does not meet complexity requirements`)